File: LeetCode Problems/1 to 100/80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.cpp
class Solution {
public:
    int removeDuplicates(vector<int> &nums) {
        map<int, int> mp;
        for (auto it: nums)
            mp[it]++;
        int sum = 0;
        nums.clear();
        for (auto it: mp) {
            sum += min(2, it.second);
            for (int i = 0; i < min(2, it.second); i++) {
                nums.push_back(it.first);
            }
        }
        return sum;
    }
};

File: LeetCode Problems/1 to 100/80. Remove Duplicates from Sorted Array II/80. Remove Duplicates from Sorted Array II.java
class Solution {
    public int removeDuplicates(int[] nums) {
        Map<Integer, Integer> map = new LinkedHashMap<>();
        for (int ele : nums) {
            map.merge(ele, 1, Integer::sum);
        }
        int k = 0;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            for (int i = 0; i < Math.min(2, entry.getValue()); i++) {
                nums[k++] = entry.getKey();
            }
        }
        return k;
    }
}
